### PR TITLE
Adding PGPLOT library

### PR DESCRIPTION
**Pull request by arahlin**
Due to lack of a configure script, this tool requires an additional patch to create a working makefile.

The patch creates a build configuration that uses the clustertools `CC` and `FC` environment variables, and assumes that X11 libraries and header files are already on the appropriate search paths.  Following the procedure in `sptTools/Makefile`, the patch also increases the number of concurrent devices to 100 from 8, and adds a compile target for `libcpgplot.so`.  Compiled libraries, header files, and shared files (`grfont.dat`, `rgb.txt`) are then copied from the build directory to the appropriate install directories under `$SROOT`.

This branch currently doesn't compile, because the `pgxwin` driver requires X11 header files to build, and these are not guaranteed to be available on a given system.

### CODE REVIEW

**Comment by briedel** [comment]
What is this needed for?

**Comment by arahlin** [comment]
This is for compiling GCP against clustertools at pole.  This is the last
remaining library that is required for GCP from the sptTools repo that
isn't already in clustertools.  I'm trying to clean up the GCP build, which
includes removing unnecessary dependencies on stuff that we don't already
have elsewhere.

**Comment by arahlin** [comment]
*Do* already have elsewhere.

**Comment by briedel** [comment]
So we dont need to put this in cvmfs?

**Comment by arahlin** [comment]
Correct.

**Comment by arahlin** [comment]
Ugh, `pgplot` is so old, I had to actually enable functionality that was completely deprecated in TK/TCL 8.6 for stuff to compile.  Tested on `sptcloud`, demos seem to work.

Since this package is contingent on having X11 headers, I could add a little test build of a dummy X11 program to determine whether this should build by default?

**Comment by briedel** [comment]
At this point I have to ask. Can be replaced with something from this decade?

**Comment by arahlin** [comment]
Eventually? Right now it's easier to spend an hour getting pgplot to
compile than days rewriting all of the visual bits in GCP to use something
else.

File: py2-v1/setup.sh [hunk 1]
```diff
@@ -58,7 +58,13 @@ if [ -z ${OpenCL} ]; then
     LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${SROOTBASE}/../distrib/OpenCL_$OS_ARCH/lib/$OS_ARCH
 fi
 
-for name in SROOTBASE SROOT SITE_CMAKE_DIR PATH MANPATH PKG_CONFIG_PATH LD_LIBRARY_PATH PYTHONPATH OS_ARCH GCC_VERSION GOTO_NUM_THREADS OPENCL_VENDOR_PATH CC CXX FC
+# PGPLOT
+if [ -f $SROOT/share/grfont.dat ]; then
+    PGPLOT_DIR=${SROOT}/share/pgplot
+    PGPLOT_DEV=/xwin
+fi
```
Comment (nwhitehorn) [inline py2-v1/setup.sh]: Why do these need to be set in the global environment?

Comment (arahlin) [reply py2-v1/setup.sh]: The documentation says these need to be set for any code to work.  The variable is used within the pgplot source code to find the grfont.dat and rgb.txt files.  Also, whoops, just noticed the file path it's checking is wrong...

**Pull request merged**